src/AzureCacheOptions.cs

```diff
@@ -76,13 +73,44 @@ public class AzureCacheOptions
         return ((double)age.Ticks / lifespan.Ticks) > .75; // Refresh if current token has exceeded 75% of its lifespan
     };
 
+    /// <summary>
+    /// Given an access token, produce the Redis user name to be used for authentication.
+    /// The default implementation extracts the 'oid' claim from the token using Microsoft.IdentityModel.JsonWebTokens.
+    /// Supply a different implementation if you need to use a different approach to determine the user name.
+    /// </summary>
+    public Func<string?, string> GetUserName = (string? token) =>
+    {
+        if (token is null)
+        {
+            throw new ArgumentNullException(nameof(token));
+        }
+
+        var jwtHandler = new JsonWebTokenHandler();
+
+        if (!jwtHandler.CanReadToken(token))
+        {
+            throw new Exception("Invalid token cannot be read");
+        }
+
+        var jwt = jwtHandler.ReadJsonWebToken(token);
+
+        if (jwt.TryGetClaim("oid", out var oid))
+        {
+            return oid.Value;
```

Comment:
I assume you have confirmed that not having `oid` will definitely be an error case, right?